src/App.jsx

```diff
@@ -14,33 +17,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const ALPHABETICALLY = 'Sort alphabetically';
+  const LENGTH = 'Sort by length';
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const getVisibleGoods = () => {
+    let visibleGoods = [...goods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (sortField) {
+      if (sortField === ALPHABETICALLY) {
+        visibleGoods = visibleGoods.sort((a, b) => a.localeCompare(b));
+      } else if (sortField === LENGTH) {
+        visibleGoods = visibleGoods.sort((a, b) => a.length - b.length);
+      }
+    }
+
+    if (isReversed) {
+      visibleGoods = visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
+
+  const applySorting = field => {
+    setSortField(field);
+  };
+
+  const resetList = () => {
+    setGoods(goodsFromServer);
+    setSortField('');
+    setIsReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const isModified = () =>
+    goods !== goodsFromServer || isReversed || sortField !== '';
```

Comment:
The comparison `goods !== goodsFromServer` will always evaluate to true because arrays are reference types in JavaScript. This means that even if the contents are the same, the references are different. Consider using a method like `JSON.stringify(goods) !== JSON.stringify(goodsFromServer)` or a utility function to compare the contents of the arrays.